curses-client.py

```diff
@@ -136,7 +138,15 @@ def get_name_from_number(number, type='long'):
         
 
 def on_receive(packet, interface):
-    global all_messages, selected_channel, channel_list
+    global all_messages, selected_channel, channel_list, packet_buffer
+
+    # update packet log
+    if display_log:
+        packet_buffer.append(packet)
+        if len(packet_buffer) > 20:
+            # trim buffer to 20 packets
+            packet_buffer = packet_buffer[-20:]
+        update_packetlog_win()
```

Comment:
update the buffer even if it's not being displayed so when you open it, it's already populated.

```suggestion
    packet_buffer.append(packet)
    if len(packet_buffer) > 20:
        # trim buffer to 20 packets
        packet_buffer = packet_buffer[-20:]
        
    if display_log:
        update_packetlog_win()
```